File: suites/02_numerical_computation/comparisons/monte_carlo_pi.go
package main
import "fmt"
func monteCarlo(samples int64) int64 {
	var inside, seed int64 = 0, 123456789
	for i := int64(0); i < samples; i++ {
		seed = ((seed * 1103515245) + 12345) % 2147483647
		x := seed % 10000
		seed = ((seed * 1103515245) + 12345) % 2147483647
		y := seed % 10000
		if (x*x)+(y*y) <= 100000000 { inside++ }
	}
	return (inside * 4000000) / samples
}
func main() { fmt.Println(monteCarlo(500000)) }

File: suites/02_numerical_computation/comparisons/monte_carlo_pi.rs
fn monte_carlo(samples: i64) -> i64 {
    let mut inside: i64 = 0;
    let mut seed: i64 = 123456789;
    for _ in 0..samples {
        seed = ((seed.wrapping_mul(1103515245)) + 12345) % 2147483647;
        let x = seed % 10000;
        seed = ((seed.wrapping_mul(1103515245)) + 12345) % 2147483647;
        let y = seed % 10000;
        if (x * x) + (y * y) <= 100000000 { inside += 1; }
    }
    (inside * 4000000) / samples
}
fn main() { println!("{}", monte_carlo(500000)); }
